apps/backoffice/src/pages/new-group-api-key.tsx

```diff
@@ -0,0 +1,171 @@
+import { Group } from "@/generated/api/Group";
+import { GroupPagination } from "@/generated/api/GroupPagination";
+import useFetch from "@/hooks/use-fetch";
+import { AppLayout, PageLayout } from "@/layouts";
+import SupervisedUserCircleIcon from "@mui/icons-material/SupervisedUserCircle";
+import { Box, Grid, Stack, Typography } from "@mui/material";
+import { useRouter } from "next/router";
+import { useSession } from "next-auth/react";
+import { useTranslation } from "next-i18next";
+import { serverSideTranslations } from "next-i18next/serverSideTranslations";
+import { ReactElement, useEffect, useState } from "react";
+import React from "react";
+import { FormProvider, useForm } from "react-hook-form";
+
+const pageTitleLocaleKey = "routes.keys.new-group-api-key.title";
+const pageDescriptionLocaleKey = "routes.keys.new-group-api-key.description";
+
+export default function NewGroupApiKey() {
+  const { t } = useTranslation();
+  const { data: session } = useSession();
+  const router = useRouter();
+
+  const showDialog = useDialog();
+  const { data: groupsData, fetchData: groupsFetchData } =
+    useFetch<GroupPagination>();
+
+  const { fetchData: subscriptionFetchData } =
+    useFetch<CreateManageGroupSubscription>();
+
+  const [formStatus, setFormStatus] = useState<boolean>(true);
+  const [selectedId, setSelectedId] = useState<string>("");
+
+  const handleFormValidation = (
+    isFormValid: boolean,
+    formSelectedId: string,
+  ) => {
+    setFormStatus(isFormValid);
+    setSelectedId(formSelectedId);
+  };
+
+  const institutionId = session?.user?.institution.id as string;
+  const groupId = selectedId;
+
+  const fetchGroups = () => {
+    groupsFetchData("getInstitutionGroups", { institutionId }, GroupPagination);
+  };
+
+  useEffect(() => {
+    fetchGroups();
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    const groupList = groupsData?.value;
+    console.log(groupList);
+  }, [groupsData]);
+
+  const methods = useForm({
+    defaultValues: {
+      group: "",
+    },
+  });
+
+  const placeholderGroups: Group[] = [
+    {
+      id: "value1",
+      name: "label1",
+    },
+    {
+      id: "value2",
+      name: "label2",
+    },
+  ];
+
+  const handleCancel = async () => {
+    const confirmed = await showDialog({
+      message: t("forms.cancel.description"),
+      title: t("forms.cancel.title"),
+    });
+    if (confirmed) {
+      console.log("operation cancelled");
+      router.back();
+    } else {
+      console.log("modal cancelled");
+    }
+  };
+
+  const handleConfirm = async () => {
+    await subscriptionFetchData(
+      "upsertManageGroupSubscription",
+      { body: { groupId } },
+      CreateManageGroupSubscription,
+      { notify: "all" },
+    );
+    router.back();
+  };
+
+  return (
+    <>
+      <PageHeader
+        description={pageDescriptionLocaleKey}
+        hideBreadcrumbs
+        onExitClick={() => router.push("/services")}
```

Comment:
Why are we navigating to "services" section? I would expect to return to the parent "keys" section